oval_graph/client.py

```diff
@@ -23,18 +21,40 @@ def __init__(self, args):
         self.source_filename = self.arg.source_filename
         self.rule_name = self.arg.rule_id
         self.out = self.arg.output
+        self.verbose = self.arg.verbose
+
+        self.parts = self.get_src('parts')
+        self.START_OF_FILE_NAME = 'graph-of-'
+        self.date = str(datetime.now().strftime("-%d_%m_%Y-%H_%M_%S"))
+        self.isatty = sys.stdout.isatty()
+
         self.all_rules = self.arg.all
         self.all_in_one = None
         self.display_html = None
-        self.isatty = sys.stdout.isatty()
+        if type(self).__name__ in ('ArfToHtml', 'JsonToHtml'):
+            self._set_attributes_for_ArfToHtml_or_JsonToHtml()
+
         self.show_failed_rules = False
         self.show_not_selected_rules = False
-        self.xml_parser = XmlParser(
-            self.source_filename)
-        self.parts = self.get_src('parts')
-        self.START_OF_FILE_NAME = 'graph-of-'
-        self.date = str(datetime.now().strftime("-%d_%m_%Y-%H_%M_%S"))
-        self.verbose = self.arg.verbose
+        if type(self).__name__ in ('ArfToHtml', 'ArfToJson'):
+            self._set_attributes_for_ArfToHtml_or_ArfToJson()
+
+        self.xml_parser = None
+        if type(self).__name__ != 'JsonToHtml':
+            self.xml_parser = XmlParser(self.source_filename)
+
+        self.html_builder = None
+        if type(self).__name__ != 'ArfToJson':
+            self.html_builder = BuilderHtmlGraph(self.parts, self.verbose)
+
+    def _set_attributes_for_ArfToHtml_or_JsonToHtml(self):
```

Comment:
Again, it would be better to have a `_set_attributes` method in different derived classes.